code/modules/mob/living/blood_types.dm

```diff
@@ -0,0 +1,224 @@
+/datum/blood_type
+	/// Displayed name of the blood type.
+	var/name = "?"
+	/// A description of the blood type.
+	var/desc
+	/// Shown color of the blood type.
+	var/color = BLOOD_COLOR_RED
+	/// Additional lightness multiplier for the blood color, useful for when the default lightness from the greyscaling doesn't cut it and you want something more vibrant.
+	/// When set, color will be transformed into a matrix with coefficients multiplied by this value
+	var/lightness_mult = null
+	/// The cached color matrix for blood with a lightness_mult. We only need to calculate this once since blood types are singletons
+	var/list/blood_color_matrix
+	/// Blood types that are safe to use with people that have this blood type (for blood transfusions)
+	var/compatible_types = list()
+	/// What reagent is represented by this blood type?
+	var/datum/reagent/reagent_type = /datum/reagent/blood
+	/// What chem is used to restore this blood type (outside of itself, of course)?
+	var/datum/reagent/restoration_chem = /datum/reagent/iron
+	/// Whether or not this blood type should create blood trails, blood sprays, etc
+	var/no_bleed_overlays
```

Comment:
If you ask me I think this should be thrown out and all bloodtypes should be able to make trails 

If there's any particularly undesirable cases it could be handled by the mob (by a trait or something, idk)